pkg/server/skydb/authinfo.go

```diff
@@ -43,45 +43,43 @@ type ProviderInfo map[string]map[string]interface{}
 // AuthInfo contains a user's information for authentication purpose
 type AuthInfo struct {
 	ID              string       `json:"_id"`
-	Username        string       `json:"username,omitempty"`
-	Email           string       `json:"email,omitempty"`
 	HashedPassword  []byte       `json:"password,omitempty"`
 	Roles           []string     `json:"roles,omitempty"`
-	Auth            ProviderInfo `json:"auth,omitempty"` // auth data for alternative methods
+	ProviderInfo    ProviderInfo `json:"provider_info,omitempty"` // auth data for alternative methods
 	TokenValidSince *time.Time   `json:"token_valid_since,omitempty"`
 	LastLoginAt     *time.Time   `json:"last_login_at,omitempty"`
 	LastSeenAt      *time.Time   `json:"last_seen_at,omitempty"`
 }
 
-// NewAuthInfo returns a new AuthInfo with specified username, email and
-// password. An UUID4 ID will be generated by the system as unique identifier
-func NewAuthInfo(username string, email string, password string) AuthInfo {
-	id := uuid.New()
+// AuthData contains the unique authentication data of a user
+// e.g.: {"username": "userA", "email": "userA@abc.com"}
+type AuthData map[string]interface{}
```

Comment:
You need a better name. `AuthData` and `AuthInfo` looks pretty similar too. People will keep asking this question and I think `AuthData` need a more descriptive name.